test/jcpsantiago/bulgogi_test.clj

```diff
@@ -1,9 +1,90 @@
 (ns jcpsantiago.bulgogi-test
   (:require
     [clojure.test :refer :all]
-    [jcpsantiago.bulgogi :refer :all]))
+    [jcpsantiago.bulgogi :as SUT]
+    [clojure.string :as s]))
 
+(defn boolean->int
+  "Cast a boolean to 1/0 integer indicator"
+  [b]
+  (when boolean? b
+        (if (true? b) 1 0)))
 
-(deftest a-test
-  (testing "FIXME, I fail."
-    (is (= 0 1))))
+
+(defn email-name
+  "Lower-cased name of an email address (the bit before @)"
+  [email]
+  (-> email
+      s/lower-case
+      (s/replace-first #"@.*" "")))
+
+
+(defn n-digits-in-email-name
+  "Number of digits in the email name"
+  [{email :email}]
+  ;; we get dependency management for free because everything is just functions
+  (->> (email-name email)
+       (re-seq #"\d")
+       count))
+
+
+(defn n-chars-in-email-mail
+  "Number of characters in the email name i.e. length of the email name"
+  [{email :email}]
+  (-> (email-name email)
+      count))
+
+
+(defn diff-eur-previous-order
+  "Difference in euros between the current order and the previous one."
+  [{current-amount :current-amount previous-amount :previous-amount}]
+  (- current-amount previous-amount))
+
+
+(defn risky-item?
+  "Boolean depending on whether an item is risky or not"
+  [{brand :brand}]
+  (->> brand
+       s/lower-case
+       (re-seq #"baz corp")
+       some?
+       boolean->int))
+
+
+(defn contains-risky-item
+  "Indicator 1/0 depending on whether a risky item is present in the cart"
+  [{items :items}]
+  (->> items
+       (map #(risky-item? %))
+       (some #(= 1 %))
+       boolean->int))
+
+(defn ^{:bulgogi/co-effect ::added} needs-co-effect
```

Comment:
this looks good 👍 I also like the fact it's using namespaced keywords, makes it clear, explicit and self-documenting; the interface is still super simple, so anyone can learn it by glancing an example without much fuss, which is perfect :)

Reply:
I wonder if it should just be called a `dep` though, because we're not using interceptors at all? This language might be alien to people unfamiliar with interceptors/re-frame

Reply:
It kind of depends, since coeffect is not a re-frame specific naming scheme. It's based on coeffect context theory. There's a great explanation here: http://tomasp.net/coeffects/
We should name it `jcpsantiago.bulgogi/coeffect` though. But if you think dependency is easier I'm not going to die on this hill :)

Reply:
that's such a cool website! I didn't read a theoretical explanation about coeffects before, thanks.
I like the name `dependency`, but at the same time `coeffect` is less handholding and people can also read the docs lol let's go with your proposal `jcpsantiago.bulgogi/coeffect`